src/metax_api/models/catalog_record_v2.py

```diff
@@ -1191,6 +1201,8 @@ def create_new_version(self):
             )
         elif not self.catalog_versions_datasets():
             raise Http400('Data catalog does not allow dataset versioning')
+        elif self.state == self.STATE_DRAFT:
+            raise Http400('Cannot create new version from draft dataset')
```

Comment:
never gets to this error when creating new version from draft dataset as 
`elif not self.catalog_versions_datasets():`
comes first and draft catalog has flag `"dataset_versioning": false`

Reply:
Draft datasets can also be part of other than draft catalog, thus making this check required.